code-dot-org/code-dot-org
# Update Chef Client Root Certificate Store - Fix Ruby/Brightbox Install

Provisioning a new system is currently failing, because the web server we install Ruby from (ppa.launchpad.net) is using an SSL/TLS certificate that is signed by a root certificate authority that is not trusted by the embedded version of Ruby in the version of the Chef Client we are using.

Our existing logic to update the Chef root certificate store does not run until **after** the system has already been provisioned as it is only invoked during the `aws/ci_build` process which updates an **existing** environment that has already been built. We need to install our preferred CA certs file immediately after installing the Chef client and before executing the Chef recipe that installs Ruby.


## Links





## Testing story

Provisioning an adhoc now succeeds. Excerpt from `/var/log/chef-bootstrap-debug.log`:

```
Recipe: cdo-ruby::brightbox
  * apt_repository[brightbox-ruby-ng] action add
    * execute[apt-cache gencaches] action nothing (skipped due to action :nothing)
    * apt_update[brightbox-ruby-ng] action nothing (skipped due to action :nothing)
    * execute[install-key 80F70E11F0F0D5F10CB20E62F5DA5F09C3173AA6] action run
      [execute] Warning: apt-key output should not be parsed (stdout is not a terminal)
                Executing: /tmp/apt-key-gpghome.CIL3K6oeum/gpg.1.sh --no-tty --recv --keyserver hkp://keyserver.ubuntu.com:80 80F70E11F0F0D5F10CB20E62F5DA5F09C3173AA6
                gpg: key F5DA5F09C3173AA6: public key "Launchpad PPA for Brightbox" imported
                gpg: Total number processed: 1
                gpg:               imported: 1
      - execute apt-key adv --no-tty --recv --keyserver hkp://keyserver.ubuntu.com:80 80F70E11F0F0D5F10CB20E62F5DA5F09C3173AA6
    * execute[apt-cache gencaches] action run
      [execute] Reading package lists...
      - execute ["apt-cache", "gencaches"]
    * file[/etc/apt/sources.list.d/brightbox-ruby-ng.list] action create
      - create new file /etc/apt/sources.list.d/brightbox-ruby-ng.list
      - update content in file /etc/apt/sources.list.d/brightbox-ruby-ng.list from none to b0a049
      --- /etc/apt/sources.list.d/brightbox-ruby-ng.list        2023-02-07 06:09:28.465553979 +0000
      +++ /etc/apt/sources.list.d/.chef-brightbox-ruby-ng20230207-30456-xkikfv.list     2023-02-07 06:09:28.465553979 +0000
      @@ -1 +1,2 @@
      +deb      http://ppa.launchpad.net/brightbox/ruby-ng/ubuntu bionic main
      - change mode from '' to '0644'
      - change owner from '' to 'root'
      - change group from '' to 'root'
    * execute[apt-cache gencaches] action run
      [execute] Reading package lists...
      - execute ["apt-cache", "gencaches"]
    * apt_update[brightbox-ruby-ng] action update
      * directory[/var/lib/apt/periodic] action create (up to date)
      * directory[/etc/apt/apt.conf.d] action create (up to date)
      * file[/etc/apt/apt.conf.d/15update-stamp] action create_if_missing (up to date)
      * execute[apt-get -q update] action run
        [execute] Hit:1 http://us-east-1.ec2.archive.ubuntu.com/ubuntu bionic InRelease
                  Hit:2 http://us-east-1.ec2.archive.ubuntu.com/ubuntu bionic-updates InRelease
                  Hit:3 http://us-east-1.ec2.archive.ubuntu.com/ubuntu bionic-backports InRelease
                  Get:4 http://ppa.launchpad.net/brightbox/ruby-ng/ubuntu bionic InRelease [20.8 kB]
                  Hit:5 http://security.ubuntu.com/ubuntu bionic-security InRelease
                  Get:6 http://ppa.launchpad.net/brightbox/ruby-ng/ubuntu bionic/main amd64 Packages [5,244 B]
                  Get:7 http://ppa.launchpad.net/brightbox/ruby-ng/ubuntu bionic/main Translation-en [2,908 B]
                  Fetched 28.9 kB in 1s (38.2 kB/s)
                  Reading package lists...
        - execute ["apt-get", "-q", "update"]
      - force update new lists of packages
```

## Deployment strategy

## Follow-up work



## Privacy



## Security



## Caching

## PR Checklist:
